addons/l10n_in/models/company.py

```diff
@@ -12,3 +11,131 @@ class ResCompany(models.Model):
+                c.country_id.code == 'IN'))
+            for company in companies_need_update_fp:
+                company._update_l10n_in_fiscal_position()
+        return res
+
+    def _get_l10n_in_fiscal_tax_vals(self, use_zero_rated_igst=False):
+        tax_types = ["sale", "purchase"]
+        gst_rates = [1, 2, 5, 12, 18, 28]
+        taxes = [
+            Command.create({'tax_src_id': f"sgst_{tax_type}_{rate}", 'tax_dest_id': f"igst_{tax_type}_{rate if not use_zero_rated_igst else 0}"})
+            for tax_type in tax_types
+            for rate in gst_rates
+        ]
+        return taxes
+
+    def _get_fiscal_position_in_intra_state_vals(self, load_template):
+        state_ids = [Command.set(self.state_id.ids)] if self.state_id else False
+        name = self.state_id and 'Within %s' % self.state_id.name or 'Intra State'
+        if load_template:
+            return {
+                'name': name,
+                'sequence': 1,
+                'auto_apply': True,
+                'state_ids': state_ids,
+                'country_id': self.env.ref('base.in').id,
+            }
+        return {
+            'name': name,
+            'state_ids': state_ids
+        }
+
+    def _get_fiscal_position_in_inter_state_vals(self, load_template):
+        if load_template:
+            return {
+                'name': 'Inter State',
+                'sequence': 2,
+                'auto_apply': True,
+                'tax_ids': self._get_l10n_in_fiscal_tax_vals(),
+                'country_id': self.env.ref('base.in').id
+            }
+        return {}
+
+    def _get_fiscal_position_in_outside_in_vals(self, load_template):
+        if load_template:
+            return {
+                'name': 'Outside India',
+                'sequence': 3,
+                'auto_apply': True,
+                'tax_ids': self._get_l10n_in_fiscal_tax_vals(),
+            }
+        return {}
+    def _get_fiscal_position_in_sez_vals(self, load_template):
+        if load_template:
+            return {
+                'name': 'Special Economic Zone (SEZ)',
+                'sequence': 4,
+                'tax_ids': self._get_l10n_in_fiscal_tax_vals(),
+            }
+        return {}
+
+    def _get_fiscal_position_in_lut_sez_vals(self, load_template):
+        if load_template:
+            return {
+                'name': 'LUT - Special Economic Zone (SEZ)',
+                'sequence': 5,
+                'tax_ids': self._get_l10n_in_fiscal_tax_vals(use_zero_rated_igst=True),
+            }
+        return {}
+    def _get_fiscal_position_in_lut_outside_in_vals(self, load_template):
+        if load_template:
+            return {
+                'name': "LUT Outside India",
+                'sequence': 6,
+                'tax_ids': self._get_l10n_in_fiscal_tax_vals(use_zero_rated_igst=True),
+            }
+        return {}
+    def _get_l10n_in_fiscal_positions_vals_by_ref(self, load_template=False):
```

Comment:
What is the use of this `load_template` param? What is the effect when it's false?

Reply:
`load_template` will be only used (True) when we are loading the templates from `template_in.py` in other cases we require to create fiscals for the branches or update it when you set the state for the first time